discovery-handlers/udev/src/discovery_handler.rs

```diff
@@ -187,9 +217,23 @@ mod tests {
         let yaml = r#"
           udevRules:
           - 'KERNEL=="video[0-9]*"'
+          permissions: rwm
         "#;
         let udev_dh_config: UdevDiscoveryDetails = deserialize_discovery_details(yaml).unwrap();
         assert_eq!(udev_dh_config.udev_rules.len(), 1);
         assert_eq!(&udev_dh_config.udev_rules[0], "KERNEL==\"video[0-9]*\"");
+        assert_eq!(&udev_dh_config.permissions, "rwm");
+    }
+
+    #[test]
+    #[should_panic]
+    fn test_deserialize_discovery_details_permissions_invalid() {
+        let yaml = r#"
+          udevRules:
+          - 'KERNEL=="video[0-9]*"'
+          permissions: xyz
+        "#;
+        let config: UdevDiscoveryDetails = deserialize_discovery_details(yaml).unwrap();
+        assert_eq!(&config.permissions, "xyz")
```

Comment:
This assertion should never be reached since the error will be thrown during deserialization. A better practice than `#[should_panic]` would be to test for the specific error:
```suggestion
        match deserialize_discovery_details::<UdevDiscoveryDetails>(yaml) {
            Ok(_) => panic!("Expected error parsing invalid permissions"),
            Err(e) => assert!(e.to_string().contains("a valid permission combination")),
        }
```

Reply:
of course...to much .Net recently ;-)